client/client.go

```diff
@@ -734,40 +734,61 @@ type (
 	WorkerVersionCapabilities = internal.WorkerVersionCapabilities
 
 	// UpdateWorkerVersioningRulesOptions is the input to [client.Client.UpdateWorkerVersioningRules].
+	//
+	// Deprecated: Build-id based versioning is deprecated in favor of worker deployment based versioning
+	//
 	// WARNING: Worker versioning is currently experimental.
-	UpdateWorkerVersioningRulesOptions = internal.UpdateWorkerVersioningRulesOptions
+	UpdateWorkerVersioningRulesOptions = internal.UpdateWorkerVersioningRulesOptions //lint:ignore SA1019 transitioning to Worker Deployments
 
 	// VersioningConflictToken is a conflict token to serialize calls to [client.Client.UpdateWorkerVersioningRules].
 	// An update with an old token fails with `serviceerror.FailedPrecondition`.
 	// The current token can be obtained with [client.Client.GetWorkerVersioningRules],
 	// or returned by a successful [client.Client.UpdateWorkerVersioningRules].
+	//
+	// Deprecated: Build-id based versioning is deprecated in favor of worker deployment based versioning
+	//
 	// WARNING: Worker versioning is currently experimental.
```

Comment:
Not sure it makes sense for an API to be both deprecated and experimental, might consider removing the Experimental tag

Reply:
True, but, I mean it kinda _is_ both? Not sure it's doing any harm this way